core/src/test/java/org/apache/iceberg/TestFastAppend.java

```diff
@@ -277,4 +278,27 @@ public void testRecoveryWithoutManifestList() {
     Assert.assertTrue("Should commit the same new manifest",
         metadata.currentSnapshot().manifests().contains(newManifest));
   }
+
+  @Test
+  public void testAppendFilesFromTable() {
```

Comment:
I think this should be in a test for `Snapshot`, not `FastAppend`. Can you start a new `TestSnapshot` suite?

Reply:
Sure, I will update it in the next commit.